src/main/java/com/ngram/app/services/TimeSeries.java

```diff
@@ -29,24 +32,26 @@ public TimeSeries() {
      */
     public TimeSeries(TimeSeries ts, int startYear, int endYear) {
         super();
-        // TODO: Fill in this constructor.
+        for(int year=startYear; year<=endYear; year++) {
+        	if(ts.containsKey(year)) { 
+        		this.put(year, ts.get(year)); 
+        	}
+        }
     }
```

Comment:
Another consideration would be to traverse the this TimeSeries directly using `Map.Entry<Integer, Double>` iterator, and only if the following condition statisfied (startYear <= `entry.getKey()` <= endYear), we do a put operation to this TimeSeries.